sky/backends/backend_utils.py

```diff
@@ -155,7 +155,11 @@
     # we need to take this field from the new yaml.
     ('provider', 'tpu_node'),
     ('provider', 'security_group', 'GroupName'),
+    ('available_node_types', 'ray.head.default', 'node_config',
+     'IamInstanceProfile'),
     ('available_node_types', 'ray.head.default', 'node_config', 'UserData'),
+    ('available_node_types', 'ray.worker.default', 'node_config',
+     'IamInstanceProfile'),
```

Comment:
I don't think we need this, as our new provisioner no longer needs to look into the worker config. Similar for the line below, let's remove the line for `ray.worker.default`, 'node_config', 'UserData'`